components/service/telemetry/src/main/java/org/mozilla/telemetry/Telemetry.java

```diff
@@ -103,11 +105,13 @@ public void run() {
                 } else if (focusEventBuilder != null) {
                     measurement = ((TelemetryEventPingBuilder) focusEventBuilder).getEventsMeasurement();
                     addedPingType = focusEventBuilder.getType();
+                } else if (pocketEventBuilder != null) {
+                    measurement = ((TelemetryPocketEventPingBuilder) pocketEventBuilder).getEventsMeasurement();
+                    addedPingType = pocketEventBuilder.getType();
```

Comment:
I missed this part before.

I think there's a problem here: As I understand it you do not want to replace the normal event ping with the new ping type. But instead add an additional ping type. Your regular events would still send a regular mobile event ping and some "pocket" events would use the new ping type.

My suggestion would be to remove this code here in `queueEvent()` and instead create an additional `queuePocketEvent()` path in the Fire TV repo (I assume in [TelemetryIntegration](https://github.com/mozilla-mobile/firefox-tv/blob/master/app/src/main/java/org/mozilla/tv/firefox/telemetry/TelemetryIntegration.kt)).

You'll either need to keep a reference to your pocket ping builder ([getting added here](https://github.com/mozilla-mobile/firefox-tv/blob/6b36f79c030b186feba68562fb999415031cae68/app/src/main/java/org/mozilla/tv/firefox/telemetry/TelemetryFactory.kt#L51-L52)) or maybe add some kind of public `getPingBuilder(type: Int): TelemetryPingBuilder?` helper to  `Telemetry`.